src/gwt/src/org/rstudio/studio/client/workbench/views/environment/dataimport/DataImport.java

```diff
@@ -1,27 +1,61 @@
+/*
+ * DataImport.java
+ *
+ * Copyright (C) 2009-16 by RStudio, Inc.
+ *
+ * Unless you have received this program directly from RStudio pursuant
+ * to the terms of a commercial license agreement with RStudio, then
+ * this program is licensed to you under the terms of version 3 of the
+ * GNU Affero General Public License. This program is distributed WITHOUT
+ * ANY EXPRESS OR IMPLIED WARRANTY, INCLUDING THOSE OF NON-INFRINGEMENT,
+ * MERCHANTABILITY OR FITNESS FOR A PARTICULAR PURPOSE. Please refer to the
+ * AGPL (http://www.gnu.org/licenses/agpl-3.0.txt) for more details.
+ *
+ */
+
 package org.rstudio.studio.client.workbench.views.environment.dataimport;
 
-import org.rstudio.core.client.widget.ThemedButton;
+import org.rstudio.core.client.widget.FileChooserTextBox;
 
 import com.google.gwt.core.client.GWT;
 import com.google.gwt.uibinder.client.UiBinder;
+import com.google.gwt.uibinder.client.UiFactory;
 import com.google.gwt.uibinder.client.UiField;
 import com.google.gwt.user.client.ui.Composite;
-import com.google.gwt.user.client.ui.TextBox;
+import com.google.gwt.user.client.ui.ListBox;
 import com.google.gwt.user.client.ui.Widget;
 
-public class DataImport extends Composite {
-
-	private static DataImportUiBinder uiBinder = GWT.create(DataImportUiBinder.class);
-
-	@UiField TextBox textResource_;
-	@UiField ThemedButton buttonBrowse_;
-	
-	interface DataImportUiBinder extends UiBinder<Widget, DataImport> {
-	}
-
-	public DataImport() {
-		initWidget(uiBinder.createAndBindUi(this));
-		setWidth("400px");
-	}
+public class DataImport extends Composite
+{
+   private static DataImportUiBinder uiBinder = GWT
+         .create(DataImportUiBinder.class);
+   
+   interface DataImportUiBinder extends UiBinder<Widget, DataImport>
+   {
+   }
 
+   public DataImport()
+   {
+      initWidget(uiBinder.createAndBindUi(this));
+      setWidth("400px");
```

Comment:
Changed. Added a helper function to DomMetrics (adjustedElementSizeToDefaultMax) to adjust to the max window size (after border adjustment), it's a bit dialog window but I think this will be good since the data preview will take most of this space.